injectproxy/routes_test.go

```diff
@@ -9,21 +10,62 @@ import (
 	"testing"
 )
 
-var mockResponse = []byte("ok")
+var okResponse = []byte(`ok`)
 
-type mockHandler struct {
-	values url.Values
+// checkQueryParameterHandler verifies that the request contains the given parameter key/value.
+func checkQueryParameterHandler(key string, value string) http.HandlerFunc {
+	return http.HandlerFunc(func(w http.ResponseWriter, req *http.Request) {
+		values, err := url.ParseQuery(req.URL.RawQuery)
+		if err != nil {
+			http.Error(w, fmt.Sprintf("unexpected error: %v", err), http.StatusInternalServerError)
+		}
+		if len(values[key]) != 1 {
```

Comment:
i guess if we don't return, `values` will be nil, no?

Reply:
what i am trying to asl, if `err != nil`, do we still need to assert these invariants?

Reply:
`values` is a `map[string][]string` so the test is mostly making sure that we don't get multiple values.

Reply:
But it should be more explicit for the reader :-)